frontend/src/views/main/project/ProjectsHome.vue

```diff
@@ -92,90 +92,96 @@
   </div>
 </template>
 
-<script lang="ts">
-import {Component, Vue, Watch} from 'vue-property-decorator';
-import {ValidationObserver} from 'vee-validate'
-import {readAllProjects, readHasAdminAccess, readUserProfile} from '@/store/main/getters';
-import {Role} from '@/enums';
-import {dispatchCreateProject, dispatchGetProjects} from '@/store/main/actions';
-import {ProjectPostDTO} from '@/generated-sources';
+<script setup lang="ts">
+import {computed, onMounted, ref, watch} from "vue";
+import {ValidationObserver} from "vee-validate";
+import {dispatchCreateProject, dispatchGetProjects} from "@/store/main/actions";
+import {readAllProjects, readHasAdminAccess, readUserProfile} from "@/store/main/getters";
+import {Role} from "@/enums";
+import {ProjectPostDTO} from "@/generated-sources";
+import {toSlug} from "@/utils";
+import store from "@/store";
+import {useRoute, useRouter} from "vue-router/composables";
 import moment from "moment";
-import {toSlug} from '@/utils';
 
-@Component
-export default class ProjectsHome extends Vue {
+const route = useRoute();
+const router = useRouter();
 
-  private openCreateDialog = false; // toggle for edit or create dialog
-  private newProjectName = "";
-  private newProjectKey = "";
-  private newProjectDesc = "";
-  private creatorFilter = 0;
-  private projectCreateError = "";
+const openCreateDialog = ref<boolean>(false); // toggle for edit or create dialog
+const newProjectName = ref<string>("");
+const newProjectKey = ref<string>("");
+const newProjectDesc = ref<string>("");
+const creatorFilter = ref<number>(0);
+const projectCreateError = ref<string>("");
 
-  $refs!: {
-    dialogForm: InstanceType<typeof ValidationObserver>;
-  };
 
-  public async mounted() {
-    const f = this.$route.query.f ? this.$route.query.f[0] || "" : ""
-    this.creatorFilter = parseInt(f) || 0;
-    await this.loadProjects();
-  }
+// template ref
+const dialogForm = ref<ValidationObserver | null>(null);
```

Comment:
This one is a bit more tricky since `ValidationObserver` is not a type but a component, so this ref should be declared as:

```
const dialogForm = ref<InstanceType<typeof ValidationObserver> | null>(null);
```